backend/src/main/java/com/github/jekattack/cornergame/game/VisitService.java

```diff
@@ -0,0 +1,66 @@
+package com.github.jekattack.cornergame.game;
+
+import com.github.jekattack.cornergame.kioskdata.Kiosk;
+import com.github.jekattack.cornergame.kioskdata.KioskRepository;
+import com.github.jekattack.cornergame.userdata.CGUser;
+import com.github.jekattack.cornergame.userdata.CGUserRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class VisitService {
+
+    private final VisitRepository visitRepository;
+    private final KioskRepository kioskRepository;
+    private final CGUserRepository cgUserRepository;
+    public void createVisit(VisitCreationData visitCreationData, String username) {
+
+        //User Coordinates
+        CGUser user = cgUserRepository.findByUsername(username).orElseThrow();
+        double userLocationLat = visitCreationData.getUserLocation().getUserLocationCoordinates().getLat();
+        double userLocationLng = visitCreationData.getUserLocation().getUserLocationCoordinates().getLng();
+
+        //Kiosk Coordinates
+        Kiosk kioskToVisit = kioskRepository.findByGooglePlacesId(visitCreationData.getGooglePlacesId()).orElseThrow();
+        double kioskToVisitLat = kioskToVisit.getKioskLocation().getLocation().getLat();
+        double kioskToVisitLng = kioskToVisit.getKioskLocation().getLocation().getLng();
+
+        //Check if User visited Kiosk within the last 24 hours
+        List<Visit> allVisitsAtKioskToVisit = Arrays.stream(visitRepository.findAllByUserId(user.getId()))
+                .filter(visit -> (visit.getGooglePlacesId()).equals(visitCreationData.getGooglePlacesId())).toList();
+        if(!allVisitsAtKioskToVisit.isEmpty()){
+            List<Date> timestamps = allVisitsAtKioskToVisit.stream().map(Visit::getTimestamp).toList();
+            Date timestampNow = Date.from(Instant.now());
+            for(Date timestamp : timestamps){
+                timestamp.setTime(timestamp.getTime() + (24*60*60*1000));
+                if(timestampNow.before(timestamp)){
+                    throw new IllegalStateException();
```

Comment:
maybe dont throw an exception in cases you absolutely expect as normal behaviour (like being drunk at a corner again without waiting a full day)
Instead just return something and return a message for the user